### PR TITLE
Update sdk to 0.16.1

### LINKED ISSUE #776: Review namespace rental fee calculation
Hey there, there was a mistake in the calculation formula. The rental fee it's calculated using the last maxDifficultyBlocks and not the last maxRollbackBlocks. Sorry for the inconvenience!

### LINKED ISSUE #779: update sdk to 0.16.1

### PR DESCRIPTION
**Pull request by domakun**
1. update sdk version to 0.16.1, fixes #779
2. update getAccount() in appWallet, use SimpleWallet.createFromDTO(this.simpleWallet) to get accounts info
3. update namespace rental fee calculation, fixes #776

### CODE REVIEW

File: src/core/model/AppWallet.ts [hunk 1]
```diff
@@ -195,10 +195,8 @@ export class AppWallet {
     }
 
     getAccount(password: Password): Account {
-        const {encryptedKey, iv} = this.simpleWallet.encryptedPrivateKey
-        const {network} = this.simpleWallet
-        const privateKey = new EncryptedPrivateKey(encryptedKey, iv).decrypt(password)
-        return Account.createFromPrivateKey(privateKey, network)
+        //@ts-ignore
+        return SimpleWallet.createFromDTO(this.simpleWallet).open(password)
```
Comment (decentraliser) [inline src/core/model/AppWallet.ts]: it should not be there, but at AppWallet initialisation in localStorage

Comment (decentraliser) [reply src/core/model/AppWallet.ts]: so, it should be called in onLogin.ts/getAccountWallets

I think we should replace that:
`return wallets.map(wallet => new AppWallet(wallet))`

By that:
`return wallets.map(AppWallet.createFromDTO)`

and then in AppWallet:
```
static createFromDTO(walletDTO) {
    return new AppWallet(
        ...,
        SimpleWallet.createFromDTO(walletDTO.simpleWallet)
    )
}
```

File: src/core/model/NetworkManager.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export class NetworkManager {
    const currentEndpoint = `${this.endpoint}`
    const heightUint = await this.chainHttp.getBlockchainHeight().toPromise()
    const height = heightUint.compact()
-   const blocksInfo = await this.blockHttp.getBlocksByHeightWithLimit(`${height}`).toPromise()
+   const blocksInfo = await this.blockHttp.getBlocksByHeightWithLimit(`${height}`,1).toPromise()
```
Comment (decentraliser) [inline src/core/model/NetworkManager.ts]: need more than that then, to the value of `maxDifficultyBlocks`

Comment (decentraliser) [reply src/core/model/NetworkManager.ts]: see 
https://github.com/nemfoundation/nem2-desktop-wallet/issues/776

File: src/core/model/AppWallet.ts (hunk 1 above)
Comment (domakun) [reply src/core/model/AppWallet.ts]: fixed

File: src/core/model/NetworkManager.ts (hunk 2 above)
Comment (domakun) [reply src/core/model/NetworkManager.ts]: fixed

File: src/core/model/AppWallet.ts [hunk 3]
```diff
@@ -56,6 +57,14 @@ export class AppWallet {
...
     numberOfMosaics: number
 
+    get publicAccount(): PublicAccount {
+      return PublicAccount.createFromPublicKey(this.publicKey, this.networkType)
+    }
+
+    static createFromDTO(wallet) {
+      return Object.assign(new AppWallet(wallet),{simpleWallet:SimpleWallet.createFromDTO(wallet.simpleWallet)})
+    }
+
```
Comment (decentraliser) [inline src/core/model/AppWallet.ts]: here we want to return an AppWallet
but it's impossible to do so with Object.assign
will push a fix

File: src/core/model/AppWallet.ts [hunk 4]
```diff
@@ -195,10 +204,8 @@ export class AppWallet {
     }
 
     getAccount(password: Password): Account {
-        const {encryptedKey, iv} = this.simpleWallet.encryptedPrivateKey
-        const {network} = this.simpleWallet
-        const privateKey = new EncryptedPrivateKey(encryptedKey, iv).decrypt(password)
-        return Account.createFromPrivateKey(privateKey, network)
+        //@ts-ignore
+        return SimpleWallet.createFromDTO(this.simpleWallet).open(password)
```
Comment (decentraliser) [inline src/core/model/AppWallet.ts]: now that we can instantiate the SimpleWallet in a clean way, we don't need this hack anymore

File: src/config/packge.ts [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-    isWin32: true
+    isWin32: false
```
Comment (evias) [inline src/config/packge.ts]: why is this being change from true to false and false to true with each SDK upgrade / release ? Please elaborate on the usage of this.

Comment (dl-nice) [reply src/config/packge.ts]: This should be a package configuration. It may be brought with you when you submit it.

Comment (dl-nice) [reply src/config/packge.ts]: This should be a package configuration. It may be brought with you when she submit it.

Comment (domakun) [reply src/config/packge.ts]: This configuration is using for switching windows or mac system in build. nem2-desktop-wallet need defferent features in these two systems, so we need update isWin32 to check if this application is running normally. In most cases, this parameter is generally false.

**Pull request merged**